test/unit/inbound/process-submitted-settlement.test.js

```diff
@@ -0,0 +1,137 @@
+         })
+       }
+  }
+})
+
+describe('process submitted settlement request', () => {
+  jest.mock('../../../app/inbound/get-existing-settlement')
+  const getExistingSettlement = require('../../../app/inbound/get-existing-settlement')
+
+  jest.mock('../../../app/inbound/save-settlement')
+  const saveSettlement = require('../../../app/inbound/save-settlement')
+
+  const processSubmittedSettlement = require('../../../app/inbound/process-submitted-settlement')
+
+  let settlement
+
+  beforeEach(() => {
+    settlement = {
+      sourceSystem: 'SITIAgri',
+      invoiceNumber: 'S123456789A123456V003',
+      frn: 1234567890,
+      currency: 'GBP',
+      value: 30000,
+      settlementDate: 'Fri Jan 21 2022 10:38:44 GMT+0000 (Greenwich Mean Time)',
+      reference: 'PY1234567',
+      settled: true
+    }
+
+    getExistingSettlement.mockResolvedValue(null)
+    saveSettlement.mockResolvedValue(null)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should call getExistingSettlement when a valid settlement is given ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(getExistingSettlement).toBeCalled()
+  })
+
+  test('should call getExistingSettlement once when a valid settlement is given ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(getExistingSettlement).toBeCalledTimes(1)
+  })
+
+  test('should call getExistingSettlement with "settlement" and "mockTransaction" when valid settlement is given ', async () => {
+    await processSubmittedSettlement(settlement)
+    expect(getExistingSettlement).toHaveBeenCalledWith(settlement, mockTransaction)
+  })
+
+  test('should log "Duplicate settlement received, skipping {duplicateSettlement.reference}" when getExistingSettlement returns a settlement ', async () => {
+    const consoleSpy = jest.spyOn(console, 'info')
+    const duplicateSettlement = settlement
+    getExistingSettlement.mockResolvedValue(duplicateSettlement)
+
+    await processSubmittedSettlement(settlement)
+
+    expect(consoleSpy).toHaveBeenCalledWith(`Duplicate settlement received, skipping ${duplicateSettlement.reference}`)
+  })
+
+  test('should throw when duplicate settlement found ', async () => {
+    getExistingSettlement.mockRejectedValue(new Error('Duplicate settlement recieved'))
+    const wrapper = async () => {
+      await processSubmittedSettlement(settlement)
+    }
+    expect(wrapper).rejects.toThrow()
+  })
+
+  test('should throw an error when duplicate settlement found ', async () => {
+    getExistingSettlement.mockRejectedValue(new Error('Duplicate settlement recieved'))
+    const wrapper = async () => {
+      await processSubmittedSettlement(settlement)
+    }
+    expect(wrapper).rejects.toThrow(Error)
+  })
+
+  test('should callmockTransaction.rollback when a duplicate settlement is found ', async () => {
+    const duplicateSettlement = settlement
+    getExistingSettlement.mockResolvedValue(duplicateSettlement)
```

Comment:
you could just have `getExistingSettlement.mockResolvedValue(settlement)` to save you 1 line :)

Reply:
Had that originaly but thought this was easier to understand. What do you think?